src/complex/universal_machine.rs

```diff
@@ -422,3 +391,34 @@ fn found_syllable(
         *syllable_serial = 1;
     }
 }
+
+fn included(i: usize, buffer: &Buffer) -> bool {
+    let glyph = &buffer.info[i];
+    if !not_standard_default_ignorable(glyph) {
+        return false;
+    }
+    if glyph.use_category() == category::ZWNJ {
+        for glyph2 in &buffer.info[i + 1..] {
+            if not_standard_default_ignorable(&glyph2) {
+                return !glyph2.is_unicode_mark();
+            }
+        }
+    }
+    true
+}
+
+fn next_glyph(p: usize, buffer: &Buffer) -> usize {
+    let mut q = p + 1;
+    while q < buffer.info.len() && !included(q, buffer) {
```

Comment:
Isn't this basically [Iterator::position](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.position). Maybe it would be shorter and more clear. 

And `rposition` below.

Just a suggestion.

Reply:
I’ve changed these to use `Iterator::find`.